### PR TITLE
add: `tailscale`

### LINKED ISSUE #4952: PacReq: tailscale
### Types

_No response_

### URL

https://github.com/tailscale/tailscale

### Building guide

https://github.com/tailscale/tailscale#building

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ndowens**
Would close #4952

### CODE REVIEW

**Review by Elsie19 — CHANGES_REQUESTED** [review]
* Styling is wrong (run pre commit).
* packagelist isn't updated.
* Title also wrong.

**Comment by ndowens** [comment]
What's the proper usage of pre-commit. When I run it just says skipped on each

**Comment by Elsie19** [comment]
Run `pre-commit -a`.

**Comment by Elsie19** [comment]
Fails on Debian stable and Ubuntu stable.

**Comment by ndowens** [comment]
> Fails on Debian stable and Ubuntu stable.

From what I read, it seems because it is using an older go version

File: packages/tailscale/tailscale.pacscript [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+repology=("project: ${name}")
+hash="c895a0f489706535ed400b0599d7d932d9eebc5f1bad2c236408a1e4b86620e7"
+
+prepare() {
+  go mod vendor
+  sed -i 's,/usr/sbin,/usr/bin/,g' cmd/tailscaled/tailscaled.service
+}
+
+build() {
+  export CGO_CPPFLAGS="${CPPFLAGS}"
```
Comment (Elsie19) [inline packages/tailscale/tailscale.pacscript:19]: these vars are empty, no?

**Review by Elsie19 — CHANGES_REQUESTED** [review]

**Review by Elsie19 — CHANGES_REQUESTED** [review]
image
Make all those fails into `incompatible` keys.

File: packages/tailscale/tailscale.pacscript [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+pkgver="1.54.0"
+url="https://github.com/tailscale/tailscale/archive/refs/tags/v${pkgver}.tar.gz"
+homepage="https://tailscale.com"
+depends=('iptables')
+makedepends=('golang-go' 'git')
+pkgdesc="Private WireGuard networks made easy"
+maintainer="Nathan <ndowens@artixlinux.org>"
+arch=('amd64')
+repology=("project: ${name}")
+incompatible=('ubuntu:latest' 'debian:stable')
```
Comment (oklopfer) [inline packages/tailscale/tailscale.pacscript:11]: `ubuntu:latest` and `debian:stable` are identifiers for the GitHub builders themselves, the actual versions of those right now are `ubuntu:jammy` and `debian:bookworm`, you should change the array to these values.

**Review by oklopfer — CHANGES_REQUESTED** [review]

**Review by oklopfer — APPROVED** [review]

**Pull request merged**